contacts/main.js

```diff
@@ -0,0 +1,566 @@
+      newLi.appendChild(fav_count);
+
+      let image = document.createElement('img');
+      image.src = item.image;
+      image.alt = 'pic';
+      image.className = 'avatar';
+      newLi.appendChild(image);
+
+      let full_name = document.createElement('span');
+      full_name.className = 'full-name';
+      let first_name = document.createElement('span');
+      first_name.innerHTML = item.firstName;
+      first_name.style.marginRight = '10px';
+      full_name.appendChild(first_name);
+      let last_name = document.createElement('span');
+      last_name.innerHTML = item.lastName;
+      full_name.appendChild(last_name);
+      newLi.appendChild(full_name);
+
+      ol.appendChild(newLi);
+    })
+  }
+}
+//-----------------------------actions----------------------------------
+function deleteAllButtonAction() {
+  if (confirm('ARE YOU SURE YOU WANT TO DO THIS ?')) {
+    deleteSelectedContacts();
+    refreshFavsList();
+    rerenderContacts();
+    resetDeleteAllButtonValueAndState();
+  } else {
+    clearSelectedBoxes();
+    resetDeleteAllButtonValueAndState();
+  }
+}
+//----------------------------------------------------------------------
+
+function deleteSelectedContacts() {
+  let personsToDelete = getSelectedContactsIDs();
+  for (let i = 0; i < personsToDelete.length; i++) {
+    deletePerson(personsToDelete[i]);
+  }
+}
+
+function getSelectedContactsIDs() {
+  // returns an array of ids
+  let returnArrayOfIDs = [];
+  let arr = document.getElementById('contacts-list').
+  getElementsByClassName('person-field');
+  for (let i = 0; i < arr.length; i++) {
+    let input = arr[i].getElementsByTagName('input')[0];
+    if (input.type === 'checkbox' && input.checked) {
+      // compute the id -> last 6 characters
+      let personId = input.id.slice(input.id.length - 6);
+      returnArrayOfIDs.push(personId);
+    }
+  }
+  return returnArrayOfIDs;
+}
+
+function clearSelectedBoxes() {
+  let ch = getSelectedContactsIDs();
+  ch.forEach((id) => {
+    let buttonId = 'checkbox-' + id;
+    document.getElementById(buttonId).checked = false;
+  })
+}
+
+///////////////////////validation functions///////////////////////
+function validateFirstNameInput(firstName) {
+  var regex = /^[A-Za-z0-9 ]+$/;
+  var isValid = regex.test(firstName);
+  if (!isValid)
+    throw new EvalError('Your first name contains special characters');
+}
+
+function validaetLastNameInput(lastName) {
+  let regex = /^[A-Za-z0-9 ]+$/;
+  let isValid = regex.test(lastName);
+  if (!isValid)
+    throw new EvalError('Your last name contains special characters');
+}
+
+function validateImageUrl(imageUrl) {
+  let pattern = /(ftp|http|https):\/\/(\w+:{0,1}\w*@)?(\S+)(:[0-9]+)?(\/|\/([\w#!:.?+=&%@!\-\/]))?/;
+  let isValid = pattern.test(imageUrl);
+  if (!isValid)
+    throw new EvalError('Your image is not an url');
+}
+/////////////////////////////////////////////////////////////////
+function getFirstNameInput() {
```

Comment:
use single method here, since they do pretty much the same thing. 

getInput would be a good name, and pass the name as a param